src/pages/payments/refund/Refund.tsx

```diff
@@ -127,12 +127,11 @@ export default function Refund() {
   }, [formik.values.invoices]);
 
   useEffect(() => {
-    if (companyGateway) {
+    if (companyGateway && payment) {
       const gateway: Gateway = companyGateway.data.data.gateway;
+      const gatewayTypeId = payment.data.data.gateway_type_id;
 
-      const showGatewayRefund = Object.values(gateway.options).some(
-        (option) => option.refund
-      );
+      const showGatewayRefund = gateway.options[gatewayTypeId].refund;
```

Comment:
Won't this crash if we are trying to access w/the index missing? Shouldn't TS complain here?

Reply:
I don't think we need to worry about the app crashing in this case. If the `Payment` object doesn't have a `company_gateway_id`, we won't even get to this part of the code. And if we do have it, we'll always be able to access the correct property under the `gateway.options` object. Still, I added an extra constraint to prevent any possibility of the app crashing.